src/sagemaker_huggingface_inference_toolkit/transformers_utils.py

```diff
@@ -273,8 +283,24 @@ def get_pipeline(task: str, device: int, model_dir: Path, **kwargs) -> Pipeline:
     else:
         kwargs["tokenizer"] = model_dir
 
-    # load pipeline
-    hf_pipeline = pipeline(task=task, model=model_dir, device=device, **kwargs)
+    if TRUST_REMOTE_CODE and os.environ.get("HF_MODEL_ID", None) is not None and device == 0:
+        torch_dtype = torch.bfloat16 if torch.cuda.get_device_capability()[0] == 8 else torch.float16
```

Comment:
This `if-elif-else` seems odd to me. Could the `if` ever match when we want to use `diffusers`?

Also, why is this extra logic in the `if` case regarding `bfloat16` and whatever else only used when `TRUST_REMOTE_CODE` is `True`, but then `TRUST_REMOTE_CODE` is also passed in other cases?

Reply:
What is odd for you? 
1. branch is when you have `TRUST_REMOTE_CODE` which is for custom modelling available through the hub and it only work if there is a `HF_MODEL_ID` and you have GPUs. -> was needed for mpt. 
2. checks whether we have diffusers installed and if we have a diffusers task and to load image generation model. 
3. is default where we try to load the custom pipeline

Reply:
It's strange because there's special bfloat16 logic but only when TRUST_REMOTE_CODE is True. I mentioned that in my comment. And it also seemed strange to me because of what I said in my other comment below.